install.sh

```diff
@@ -902,7 +898,7 @@ case $platform in
     filetype="rpm"
     platform_package="el"
     # For Amazon Linux 2023 and onwards we can use the 'amazon' packages created instead of 'el' packages
-    if (( $platform_version >= 2 )); then
+    if (( $platform_version == 2023 || $platform_version == 2 )); then
```

Comment:
Thanks @joshcooper,
Those changes are being pulled from https://github.com/puppetlabs/puppetlabs-puppet_agent/blob/cae24809600cf8768f769ea228edfb3ffea69014/tasks/install_shell.sh#L666
Since, this PR is for Ubuntu-24, I have removed the change. 

I have raised this ticket to separately fix this in puppet_agent module and then update here: https://perforce.atlassian.net/browse/PA-6677